src/order/well_founded_set.lean

```diff
@@ -202,6 +206,53 @@ theorem partially_well_ordered_on.image_of_monotone_on {s : set α}
     exact (classical.some_spec (h n)).1 }
 end
 
+lemma _root_.is_antichain.finite {s : set α} {r : α → α → Prop} (ha : is_antichain r s)
```

Comment:
`finite_of_partially_well_ordered_on`? It's longer but we might have more lemmas claiming that an antichain is finite under some additional conditions (e.g., `is_antichain.finite_of_is_pwo` to the case `r = (≤)`).

Reply:
Sure